### DISCUSSION TITLE
Where to copy "email-code-form.ftl"

### DISCUSSION THREAD

**Discussion by fatehali429** [mesutpiskin/keycloak-2fa-email-authenticator#11] [Q&A] [answered]
In documentation i don't see the location of email-code-form.ftl file.

I am able to get email with OTP but don't see the Form in browser where i suppose to enter OTP.

**Comment by mfernandezmsistemas** [comment]
I have the same problem. Sending of Email perfect, but then I get an error and it seems to be because of not showing the customized form with more buttons that the MFA App standard flow to insert the code from the app (like Resend button). 

I have the email-code-form.ftl but I don´t know where to place it.

I tried to build a new Theme inside "keycloak/themes", exatly in the new theme to de login part keycloak\themes\email-code-theme\login  but I get the same error and I´m not sure if I´m right with the place.

**Reply by ghciotti** [reply]
I would check your POM file and make sure the versioning is correct, the  email-code-form.ftl was not needed in my deployment to get it to work, I think it is included in the compiled jar file that is uploaded to the server, but I was getting an error with it initially and recompiled the code with some updated versioning and it worked

**Comment by AnggitHerdianto** [comment, marked as answer]
Here is the resource : 

https://github.com/mesutpiskin/keycloak-2fa-email-authenticator/tree/main/src/main/resources/theme/email-code-theme/email

I use docker, and this one is working for me :

![Screenshot from 2024-10-18 10-53-58](https://github.com/user-attachments/assets/b67f934e-92fd-454c-9d1f-9d2b18411a6c)
![Screenshot from 2024-10-18 10-53-28](https://github.com/user-attachments/assets/49a4f5fc-d16c-4dec-bb06-2248ed53f880)
